easybuild/easyconfigs/p/Python/Python-3.6.3-icc_segfault_workaround.patch

```diff
@@ -0,0 +1,65 @@
+This patch was provided by Andrew Edmondson in an email on the EasyBuild mailing list February 28, 2018:  https://lists.ugent.be/wws/arc/easybuild/2018-02/msg00043.html
+
+The patch does two thing:
+1. Expands a macro explicitly to avoid generating an overly long line.
+2. Compiles that file with -O0
+
+
+--- ./Modules/_sha3/kcp/KeccakP-1600-opt64.c.orig	2018-02-26 16:29:31.750301802 +0000
++++ ./Modules/_sha3/kcp/KeccakP-1600-opt64.c	2018-02-26 17:27:32.243042188 +0000
+@@ -463,12 +463,44 @@
+     UINT64 *inDataAsLanes = (UINT64*)data;
+ 
+     copyFromState(A, stateAsLanes)
++
+     while(dataByteLen >= laneCount*8) {
+         addInput(A, inDataAsLanes, laneCount)
+-        rounds24
++        //rounds24
++        // ACE 26/02/18 - icc segfaults on this - presumably because the line is too long for its buffer
++        // So the following is the output of the macro (with full unrolling enabled)
++        // start
++        prepareTheta
++        thetaRhoPiChiIotaPrepareTheta( 0, A, E)
++        thetaRhoPiChiIotaPrepareTheta( 1, E, A)
++        thetaRhoPiChiIotaPrepareTheta( 2, A, E)
++        thetaRhoPiChiIotaPrepareTheta( 3, E, A)
++        thetaRhoPiChiIotaPrepareTheta( 4, A, E)
++        thetaRhoPiChiIotaPrepareTheta( 5, E, A)
++        thetaRhoPiChiIotaPrepareTheta( 6, A, E)
++        thetaRhoPiChiIotaPrepareTheta( 7, E, A)
++        thetaRhoPiChiIotaPrepareTheta( 8, A, E)
++        thetaRhoPiChiIotaPrepareTheta( 9, E, A)
++        thetaRhoPiChiIotaPrepareTheta(10, A, E)
++        thetaRhoPiChiIotaPrepareTheta(11, E, A)
++        thetaRhoPiChiIotaPrepareTheta(12, A, E)
++        thetaRhoPiChiIotaPrepareTheta(13, E, A)
++        thetaRhoPiChiIotaPrepareTheta(14, A, E)
++        thetaRhoPiChiIotaPrepareTheta(15, E, A)
++        thetaRhoPiChiIotaPrepareTheta(16, A, E)
++        thetaRhoPiChiIotaPrepareTheta(17, E, A)
++        thetaRhoPiChiIotaPrepareTheta(18, A, E)
++        thetaRhoPiChiIotaPrepareTheta(19, E, A)
++        thetaRhoPiChiIotaPrepareTheta(20, A, E)
++        thetaRhoPiChiIotaPrepareTheta(21, E, A)
++        thetaRhoPiChiIotaPrepareTheta(22, A, E)
++        thetaRhoPiChiIota(23, E, A)
++        // end
++
+         inDataAsLanes += laneCount;
+         dataByteLen -= laneCount*8;
+     }
+     copyToState(stateAsLanes, A)
++
+     return originalDataByteLen - dataByteLen;
+ }
+--- setup.py.orig	2018-02-26 22:53:30.915149439 +0000
++++ setup.py	2018-02-26 22:53:01.409070915 +0000
+@@ -916,6 +916,7 @@
+         sha3_deps.append('hashlib.h')
+         exts.append( Extension('_sha3',
+                                ['_sha3/sha3module.c'],
++                               extra_compile_args=['-O0'],
```

Comment:
@edmondac Which other options did you try here? `-O1`? `-O2` with specific optimizations disabled to try and dance around the problem?

I agree with the upstream concern that this severely limits the applicability of this patch, it really cripples the Python installation...

On which OSs are you seeing the segmentation fault? I'm wondering why I haven't seen this at all...

Does setting `ulimit -s unlimited` help by any chance to fix the build (without using this patch)?

Reply:
@boegel Why does this cripple the Python installation?  It is only the SHA3 hash module that is compiled with -O0, I guess it is rare that an application relies on performance of sha3 (a part from cracking passwords and perhaps mining cryptocurrencies, neither of which is usually done in Python).

Reply:
As mentioned in https://bugs.python.org/issue33174, this will make the `sha3` support very slow.

I agree that this may not be very relevant to scientific software, but I'm not sure that's true or how to check that...

On the other hand, I do not see a good alternative either (although I do keep wondering why we haven't seen this issue at all).

@schiotz @vanzod On which OSs (version) have you run into the compilation problem without this patch?

Reply:
Turns out `ulimit -s unlimited` is the reason I wasn't seeing this, let's follow up on this in #6484